bin/render-activity-stream-html.js

```diff
@@ -0,0 +1,59 @@
+const fs = require("fs");
+const path = require("path");
+const prerender = require("./prerender");
+
+const BASE_FILE_PATH = path.resolve(__dirname, "../system-addon/data/content/");
+const PRERENDERED_HTML_FILE_PATH = path.join(BASE_FILE_PATH, "activity-stream-prerendered.html");
+const HTML_FILE_PATH = path.join(BASE_FILE_PATH, "activity-stream.html");
+const INITIAL_STATE_JS_FILE_PATH = path.join(BASE_FILE_PATH, "activity-stream-initial-state.js");
+
+const DEFAULT_OPTIONS = {
+  baseUrl: "resource://activity-stream/",
+  locale: "", // TODO: pass in options.locale. For now, we're just doing empty strings.
+  direction: "ltr",
+  title: "New Tab"
+};
+
+function templateHTML(options, html) {
+  const isPrerendered = !!html;
+  return `<!doctype html>
+<html lang="${options.locale}" dir="${options.direction}">
+  <head>
+    <meta charset="utf-8">
+    <meta http-equiv="Content-Security-Policy-Report-Only" content="script-src 'unsafe-inline'; img-src http: https: data: blob:; style-src 'unsafe-inline'; child-src 'none'; object-src 'none'; report-uri https://tiles.services.mozilla.com/v4/links/activity-stream/csp">
+    <title>${options.title}</title>
+    <link rel="icon" type="image/png" id="favicon" href="chrome://branding/content/icon32.png"/>
+    <link rel="stylesheet" href="chrome://browser/content/contentSearchUI.css" />
+    <link rel="stylesheet" href="${options.baseUrl}data/content/activity-stream.css" />
+    <link rel="icon" type="image/svg+xml" href="${options.baseUrl}img/newtab-icon.svg">
+  </head>
+  <body>
+    <div id="root">${isPrerendered ? html : ""}</div>${isPrerendered ? `
+    <script src="${options.baseUrl}data/content/activity-stream-initial-state.js"></script>` : ""}
+    <script src="chrome://browser/content/contentSearchUI.js"></script>
+    <script src="${options.baseUrl}vendor/react.js"></script>
+    <script src="${options.baseUrl}vendor/react-dom.js"></script>
+    <script src="${options.baseUrl}vendor/react-intl.js"></script>
+    <script src="${options.baseUrl}vendor/redux.js"></script>
+    <script src="${options.baseUrl}vendor/react-redux.js"></script>
+    <script src="${options.baseUrl}data/content/activity-stream.bundle.js"></script>
+  </body>
+</html>
+`;
+}
+
+function templateJs(state) {
+  return `// Note - this is a generated file.
+  window.gActivityStreamPrerenderedState = ${JSON.stringify(state, null, 2)};
+`;
+}
+
+const args = require("minimist")(process.argv.slice(2), {alias: {baseUrl: "b", title: "t", locale: "l", direction: "d"}});
+const options = Object.assign({}, DEFAULT_OPTIONS, args || {});
+const {html, state} = prerender(options.locale);
+
+fs.writeFileSync(HTML_FILE_PATH, templateHTML(options)); // eslint-disable-line no-sync
+fs.writeFileSync(PRERENDERED_HTML_FILE_PATH, templateHTML(options, html)); // eslint-disable-line no-sync
+fs.writeFileSync(INITIAL_STATE_JS_FILE_PATH, templateJs(state)); // eslint-disable-line no-sync
+
```

Comment:
It'd improve readaibility to just put /* eslint-disable no-sync */ at the top of the file.